bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java

```diff
@@ -0,0 +1,191 @@
+    keySplits = new byte[keys.length][];
+    for (int i = 0; i < keys.length; i++) {
+      keySplits[i] = keys[i].getBytes();
+    }
+  }
+
+  private static String getTestProperty(String name) {
+    return checkNotNull(System.getProperty(name), "Required property missing: " + name);
+  }
+
+  @After
+  public void teardown() throws IOException {
+    final List<GcsPath> paths = gcsUtil.expand(GcsPath.fromUri(restoreDir + "/*"));
+
+    if (!paths.isEmpty()) {
+      final List<String> pathStrs = new ArrayList<>();
+
+      for (GcsPath path : paths) {
+        pathStrs.add(path.toString());
+      }
+      this.gcsUtil.remove(pathStrs);
+    }
+
+    connection.close();
+
+    // delete test table
+    BigtableConfiguration.connect(projectId, instanceId)
+        .getAdmin()
+        .deleteTable(TableName.valueOf(tableId));
+  }
+
+  @Test
+  public void testHBaseSnapshotImport() throws Exception {
+
+    try (Connection connection = BigtableConfiguration.connect(projectId, instanceId)) {
+      // Crete table
+      System.out.println("DEBUG (create test table) ==>");
+      TableName tableName = TableName.valueOf(tableId);
+      HTableDescriptor descriptor = new HTableDescriptor(tableName);
+
+      descriptor.addFamily(new HColumnDescriptor(CF));
+
+      connection.getAdmin().createTable(descriptor, SnapshotTestingUtils.getSplitKeys());
+
+      // Start import
+      System.out.println("DEBUG (import snapshot) ==>");
+      DataflowPipelineOptions importPipelineOpts =
+          PipelineOptionsFactory.as(DataflowPipelineOptions.class);
+      importPipelineOpts.setRunner(DirectRunner.class);
+      importPipelineOpts.setGcpTempLocation(dataflowStagingLocation);
+      importPipelineOpts.setNumWorkers(1);
+      importPipelineOpts.setProject(projectId);
+
+      ImportJobFromHbaseSnapshot.ImportOptions importOpts =
+          importPipelineOpts.as(ImportJobFromHbaseSnapshot.ImportOptions.class);
+      // setup GCP and bigtable
+      importOpts.setBigtableProject(StaticValueProvider.of(projectId));
+      importOpts.setBigtableInstanceId(StaticValueProvider.of(instanceId));
+      importOpts.setBigtableTableId(StaticValueProvider.of(tableId));
+      importOpts.setBigtableAppProfileId(null);
+
+      // setup Hbase snapshot info
+      importOpts.setHbaseRootDir(StaticValueProvider.of(hbaseSnapshotDir));
+      importOpts.setRestoreDir(StaticValueProvider.of(restoreDir));
+      importOpts.setSnapshotName(StaticValueProvider.of(TEST_SNAPSHOT_NAME));
+
+      // run pipeline
+      State state = ImportJobFromHbaseSnapshot.buildPipeline(importOpts).run().waitUntilFinish();
+      Assert.assertEquals(State.DONE, state);
+
+      // check data in bigtable
+      BigtableTableUtils destTable = new BigtableTableUtils(connection, tableId, CF);
+      Assert.assertEquals(
+          100 /* There are 100 rows in test snapshot*/,
+          destTable.readAllCellsFromTable().toArray().length);
```

Comment:
Please add a TODO for me, we will add the sync-table after this. That way the data in this table can be verified.